include/erlcloud_aws.hrl

```diff
@@ -137,6 +137,11 @@
           %% If you provide a custom function be aware of this anticipated change.
           %% See erlcloud_retry for full documentation.
           retry=fun erlcloud_retry:no_retry/1::erlcloud_retry:retry_fun(),
+
+          %% By default treat all non 2xx http error codes as errors.
+          %% But in some cases, like lambda call it useful to override such
+          %% behaviour by custom one.
+          response_type=fun erlcloud_retry:only_http_errors/1::erlcloud_retry:response_type_fun(),
```

Comment:
I wouldn't complicate retry logic with yet another flag, but instead I suggest we add a new helper retry function `erlcloud_retry:lambda_retry()` which pays attention to `x-amz-function-error` header and retries it. Essentially in this change rename `aws_api_errors` to `lambda_retry()` and remove everything else.

Reply:
@kkuzmin in current implementation we can't handle `x-amz-function-error` in retry, because response is marked as 'ok' and retry function wouldn't be called at all.
So the only solution is to allow to mark such responses as errors by introduction new 'response_type' callback.
It gives as flexibility without any breaking changes.

Reply:
I followed the code path and that makes sense now. Thank you.
In this case I think `retry_response_type` looks clearer than `response_type` in `#aws_config{}`. Could you please rename it.